assets/src/blocks/Cookies/useCookie.js

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from 'react';
+
+export const readCookie = (name) => {
+  const declarations = document.cookie.split(';');
+  let match = null;
+  declarations.forEach(part => {
+    const [key, value] = part.split('=');
+    if (key.trim() === name) {
+      match = value;
+    }
+  });
+  return match;
+};
+
+export const writeCookie = (name, value, days = 365) => {
+  let date = new Date();
+  date.setTime(date.getTime() + (days * 24 * 60 * 60 * 1000));
+  let secureMode = document.location.protocol === 'http:'
+    ? ';SameSite=Lax'
+    : ';SameSite=None;Secure';
+  document.cookie = encodeURI(name) + '=' + encodeURI(value) + ';domain=.' + document.domain + ';path=/;' + '; expires=' + date.toGMTString() + secureMode;
+};
+
+// Value should not matter as cookie is expired.
+export const removeCookie = name => writeCookie(name, '0', -1);
+
+export const useCookie = name => {
+  const [value, setValue] = useState(() => readCookie(name));
+
+  const saveCookie = () => {
+    if (value === null) {
+      removeCookie(name);
+      return;
+    }
+    writeCookie(name, value);
+  };
+  useEffect(saveCookie, [value]);
+
+  return [
```

Comment:
I think it would be clearer to return an object here instead of an array? 🤔 Then we can label what's returned, and also we don't need to worry about which order we write them in when using the hook.

Reply:
There's a good reason for using arrays, which is that most of the time the name of object properties will be too generic. Most usages of the hook will need to give it a more specific name. Imagine having to do 
```
const { value: myFooValue, setValue: setMyFooValue) = useState('defaultFoo')
```
every time you want to use state. Many custom hooks will suffer from this issue too, and in this case it also needs to be called specifically.

That said I do agree that for more than 2 things being returned it maybe becomes too error prone.

Reply:
Fair point, I guess it's also fine here to use an array since this hook is not meant to be reused 👍